corehq/apps/userreports/sql/adapter.py

```diff
@@ -44,30 +45,40 @@ def get_query_object(self):
         """
         return self.session_helper.Session.query(self.get_table())
 
+    def best_effort_save(self, doc):
+        """
+        Does a best-effort save of the document. Will fail silently if the save is not successful.
+
+        For certain known, expected errors this will do no additional logging.
+        For unexpected errors it will log them.
+        """
+        try:
+            self.save(doc)
```

Comment:
Based on your point that if there's an `IntegrityError` then future saves will also fail I decided to just remove requirement 4 from the previous thread and see if it comes up.